trieste/utils/objectives.py

```diff
@@ -126,6 +126,103 @@ def logarithmic_goldstein_price(x: TensorType) -> TensorType:
 """
 
 
+def hartmann_3(x: TensorType) -> TensorType:
+    """
+    The Hartmann 3 test function over :math:`[0, 1]^3`. This function has 3 local
+    and one global minima.
+
+    :param x: The points at which to evaluate the function, with shape [..., 3].
+    :return: The function values at ``x``, with shape [..., 1].
+    :raise ValueError (or InvalidArgumentError): If ``x`` has an invalid shape.
+    """
+    tf.debugging.assert_shapes([(x, (..., 3))])
+
+    a = tf.constant([1.0, 1.2, 3.0, 3.2], dtype=tf.float64)
+    A = tf.constant(
+        [[3.0, 10.0, 30.0], [0.1, 10.0, 35.0], [3.0, 10.0, 30.0], [0.1, 10.0, 35.0]],
+        dtype=tf.float64,
+    )
+    P = tf.constant(
+        [
+            [3689.0, 1170.0, 2673.0],
+            [4699.0, 4387.0, 7470.0],
+            [1091.0, 8732.0, 5547.0],
+            [381.0, 5743.0, 8828.0],
+        ],
+        dtype=tf.float64,
+    )
+    P = P * 10 ** (-4)
+
+    inner_sum = -tf.reduce_sum(A * tf.math.square(tf.expand_dims(x, 1) - P), -1)
+    return -tf.reduce_sum(a * tf.math.exp(inner_sum), -1)
+
+
+HARTMANN_3_MINIMIZER = tf.constant([[0.114614, 0.555649, 0.852547]], tf.float64)
+"""
+The global minimizer for the :func:`hartmann_3` function, with shape [1, 3] and
+dtype float64.
+"""
+
+
+HARTMANN_3_MINIMUM = tf.constant([-3.86278], tf.float64)
+"""
+The global minimum for the :func:`hartmann_3` function, with shape [1] and dtype
+float64.
+"""
+
+
+def hartmann_6(x: TensorType) -> TensorType:
+    """
+    The Hartmann 6 test function over :math:`[0, 1]^6`. This function has
+    6 local and one global minima.
```

Comment:
there are references for this in https://www.sfu.ca/~ssurjano/hart6.html

do you think it's worth adding one?

Reply:
Can I add just the URL or do you want a proper reference in the docs file?

Reply:
no need to add the URL - that's already in the module docstring. I was thinking a proper reference, but I'll leave this up to you. I'm pretty ambivalent